src/main.rs

```diff
@@ -66,36 +60,37 @@ Notes:
   If your door does not use DOOR32.SYS you can omit --dropfile and --out and still
   use {fd}.";
 
-const DOOR32_SYS_FILENAME: &'static str = "DOOR32.SYS";
+const DOOR32_SYS_FILENAME: &str = "DOOR32.SYS";
 
 #[derive(Debug, Deserialize)]
 struct Args {
-    flag_port:      i32,
-    flag_dropfile:  String,
-    flag_out:       String,
-    arg_target:     String,   
-    flag_version:   bool
+    flag_port: i32,
+    flag_dropfile: String,
+    flag_out: String,
+    arg_target: String,
+    flag_version: bool,
 }
 
 //
 //  DOOR32.SYS
 //  https://github.com/NuSkooler/ansi-bbs/blob/master/docs/dropfile_formats/door32_sys.txt
 //
-fn read_door32sys_dropfile(dropfile_path: &str) -> Result<String, String> {
-    let path = Path::new(dropfile_path);
-    let is_door32_filename = match path.file_name() {
-        Some(ref f) => f.to_string_lossy().eq_ignore_ascii_case(DOOR32_SYS_FILENAME),
-        None        => false,
-    };
-
-    if !is_door32_filename {        
-        return Err(format!("File at {} does not appear to be DOOR32.SYS", dropfile_path));
-    }
-
-    match fs::read(path) {
-        Ok(contents)    => Ok(String::from_cp437(contents, &CP437_CONTROL)),
-        Err(e)          => Err(e.to_string()),
+fn read_door32sys_dropfile(dropfile_path: &str) -> Result<String, Error> {
+    let pathname = Path::new(dropfile_path).file_name().ok_or(Error::new(
+        ErrorKind::NotFound,
+        format!("File {} does not name a file", dropfile_path),
+    ))?;
+    if !pathname
+        .to_string_lossy()
+        .eq_ignore_ascii_case(DOOR32_SYS_FILENAME)
+    {
+        return Err(Error::new(
+            ErrorKind::InvalidInput,
```

Comment:
Ah good, this is exactly what I wanted actually. I think some of the code could use the shorthand `?` stuff now that Error's are returned.

Reply:
Yup. You can `?` pretty much all the way up to `fn main`.